src/digital_audio_path.rs

```diff
@@ -61,12 +61,12 @@ pub struct DigitalAudioPath {
 impl DigitalAudioPath {
     pub fn new() -> Self {
         DigitalAudioPath {
-            data: 0b0_0000_0000,
+            data: 0b0_0000_1000,
         }
     }
 
-    /// ADC high pass filter
-    pub fn adc_hpf(&mut self) -> EnableDisable {
+    /// Disable ADC high pass filter. Enabling this disable the high pass filter
+    pub fn adc_hpf_disable(&mut self) -> EnableDisable {
```

Comment:
Great catch here, this was a huge oversight on my end!

I find the "enable to disable" interface here very confusing, though, and I know as an end-user this would trip me up. My suggestion here would be to introduce a new struct that is the inverse of `EnableDisable` specifically for this, so the interface remains `adc_hpf.enable()` or `adc_hpf.disable()`.

Something like this:

```rs
pub struct EnableDisable<'a> {
    index: u16,
    bitmask: BitMask<'a>,
}

impl<'a> AdcHpf<'a> {
    pub fn new(index: u16, data: &'a mut u16) -> Self {
        let bitmask = BitMask::new(data);

        AdcHpf { index, bitmask }
    }

    pub fn enable(&mut self) {
        self.bitmask.unset(self.index);
    }

    pub fn disable(&mut self) {
        self.bitmask.set(self.index);
    }
}
```

Happy to make this change on my own if you'd prefer!

Reply:
here, it depend how much abstraction you want. In more general case, I frequently saw this kind of confusion on PAC, so i think it's wise to put indication of what happen under the hood if you invert the EnableDisable.

Reply:
done